api/buckets/service.go

```diff
@@ -2131,18 +2213,50 @@ func (s *Service) Archive(ctx context.Context, req *pb.ArchiveRequest) (*pb.Arch
 				return nil, fmt.Errorf("there is an in progress archive")
 			// Case 1.c.
 			case ffs.Failed, ffs.Canceled:
-				jid, err = s.PGClient.FFS.PushStorageConfig(ctxFFS, p.Cid(), powc.WithOverride(true))
+				jid, err = s.PGClient.FFS.PushStorageConfig(ctxFFS, p.Cid(), powc.WithStorageConfig(storageConfig), powc.WithOverride(true))
 				if err != nil {
 					return nil, fmt.Errorf("pushing config: %s", err)
 				}
 			default:
 				return nil, fmt.Errorf("unexpected current archive status: %d", ba.Archives.Current.JobStatus)
 			}
 		} else { // Case 2.
-			jid, err = s.PGClient.FFS.Replace(ctxFFS, oldCid, p.Cid())
+			res, err := s.PGClient.FFS.GetStorageConfig(ctxFFS, oldCid)
 			if err != nil {
-				return nil, fmt.Errorf("replacing cid: %s", err)
+				return nil, fmt.Errorf("looking up old storage config: %s", err)
+			}
+			// ToDo: Remove this once .PGClient.FFS.GetStorageConfig returns a ffs.StorageConfig
+			var oldStorageConfig *ffs.StorageConfig
+			if res.Config != nil {
+				oldStorageConfig = &ffs.StorageConfig{
+					Repairable: res.Config.Repairable,
+					Cold:       fromRPCColdConfig(res.Config.Cold),
+					Hot:        fromRPCHotConfig(res.Config.Hot),
+				}
+			}
+
+			if reflect.DeepEqual(&storageConfig, oldStorageConfig) {
+				// Old storage config is the same as the new so use replace.
+				jid, err = s.PGClient.FFS.Replace(ctxFFS, oldCid, p.Cid())
+				if err != nil {
+					return nil, fmt.Errorf("replacing cid: %s", err)
+				}
+			} else {
+				// New storage config, so remove and push.
+				_, err = s.PGClient.FFS.PushStorageConfig(ctxFFS, oldCid, powc.WithStorageConfig(ffs.StorageConfig{}), powc.WithOverride(true))
```

Comment:
@jsign I just want to check that this is ok. I have to disable hot and cold storage before calling `Remove`. I do that here, just ignore the jobId, and immediately call `Remove`. This works when I run the tests, but just want to make sure it's not just luck. Do I need to watch the job and wait until it is successful before calling `Remove`?

Reply:
It is disabled once this PR is merged in. But for previously archived buckets that are archiving again once this PR is merged (the situation we discuss above), hot storage may be enabled and will be disabled by this `PushStorageConfig`